### PR TITLE
Feature/tech 894 account for concurrent runs

### PR DESCRIPTION
**Pull request by Jorg88**

### CODE REVIEW

File: mpyl-reporter.py
Comment (Jorg88) [inline mpyl-reporter.py:1]: The changes to this file also need to be applied to `deployment/pipeline/mpyl/mpyl-post-run.py` in the monorepo after releasing this

File: src/mpyl/stages/discovery.py [hunk 1]
```diff
@@ -204,7 +204,7 @@ def find_build_set(  # pylint: disable=too-many-arguments, too-many-locals
 
     build_set: dict[Stage, set[ProjectExecution]] = {}
 
-    build_set_file = Path(BUILD_ARTIFACTS_FOLDER) / "build_plan"
+    build_set_file = Path(BUILD_ARTIFACTS_FOLDER) / "build_plan.pickle"
```
Comment (ptab) [inline src/mpyl/stages/discovery.py:207]: No problem with this change, just a reminder that we'll need to change [the monorepo workflow](https://github.com/Vandebron/Vandebron/blob/1c8c48a6e861ef33e17e4cb99a29989ae9ed19d0/.github/workflows/mpyl-pull-request.yml#L77) to publish this new filename.

File: mpyl-reporter.py [hunk 2]
```diff
@@ -20,55 +20,60 @@
...
-        config = parse_config("mpyl_config.yml")
-        properties = parse_config("run_properties.yml")
-        run_properties = construct_run_properties(config=config, properties=properties)
-
-        commit_check = CommitCheck(config=config, logger=logger)
-        accumulator.add(commit_check.send_report(run_result))
+    config = parse_config("mpyl_config.yml")
+    properties = parse_config("run_properties.yml")
+    run_properties = construct_run_properties(config=config, properties=properties)
+    run_result: RunResult = RunResult(run_properties=run_properties)
```
Comment (ptab) [inline mpyl-reporter.py:33]: For my understanding: what's the purpose of `ReportAccumulator` now that we've done this change? It looks like we're accumulating all `run_resuilt-*.pickle` files into a single `RunResult`, and then accumulating this again into an `ReportAccumulator`. Is there an opportunity to simplify this setup?

File: src/mpyl/stages/discovery.py (hunk 1 above)
Comment (Jorg88) [reply src/mpyl/stages/discovery.py:207]: I wanted to make it more explicit which file this is now that we have the same file in `.json` format as well

File: mpyl-reporter.py (hunk 2 above)
Comment (ptab) [reply mpyl-reporter.py:33]: Yeah that makes sense, thanks for explaining! 👍

**Review by ptab — APPROVED** [review]

**Review by koeves — COMMENTED** [review]
Changes look good.

So we don't forget, I would use the github run id instead of a random uuid if it's not hard to supply it

**Review by koeves — APPROVED** [review]

**Pull request merged**